projects/plugins/crm/includes/ZeroBSCRM.DAL3.Obj.Invoices.php

```diff
@@ -2169,182 +2169,169 @@ private function tidy_invoice($obj=false,$withCustomFields=false){
 
     }
 
-    /**
-     * Takes whatever invoice data available and re-calculates net, total, tax etc. 
-     * .. returning same obj with updated vals
-     * .. This is a counter to the js func which does this in-UI, so changes need to be replicated in either or
-     *
-     * @param array $invoiceData
-     *
-     * @return array $invoiceData
-     */
-    public function recalculate($invoiceData=false){
-
-        if (is_array($invoiceData)){
-
-            global $zbs;
-
-            // we pass any discount saved against main invoice DOWN to the lineitems, first:
-            if (isset($invoiceData['lineitems']) && is_array($invoiceData['lineitems'])){
-
-                // if not discounted, still recalc net
-                if (!isset($invoiceData['discount']) && $invoiceData['discount'] <= 0){
-
-                    // recalc line items, but no discount to apply
-                    foreach ($invoiceData['lineitems'] as $lineItem) { 
-
-                        $finalLineItems[] = $zbs->DAL->lineitems->recalculate($lineItem);
```

Comment:
Interesting...looks like even without defining it it works. But best to explicitly define it like you have in this PR.